modules/sov-ibc-transfer/src/context.rs

```diff
@@ -146,30 +146,27 @@ where
         coin: &PrefixedCoin,
         memo: &Memo,
     ) -> Result<(), TokenTransferError> {
-        let sov_memo: SovereignMemo = serde_json::from_str(memo.as_ref())
-            .map_err(|e| TokenTransferError::Other(format!("Failed to decode memo: {e}")))?;
+        if !memo.as_ref().is_empty() {
+            return Err(TokenTransferError::Other(
+                "Memo must be empty when burning tokens".to_string(),
+            ));
+        }
```

Comment:
Can you please explain why this is required?

Reply:
Just a safety check and disallowing someone to pass a huge memo that might overload the system. This perhaps should be taken care at ibc-rs, then we can drop it here.

Reply:
Can you please add some comments about this situation?

Reply:
[e9651e1](https://github.com/informalsystems/sovereign-ibc/pull/133/commits/e9651e12c50544acd44a1d4ce4455c65911aebba)